extensions/wikia/CreateNewWiki/tasks/CreateTables.php

```diff
@@ -55,21 +55,49 @@ public function run() {
 		$dbw = wfGetDB( DB_MASTER, [ ], $this->taskContext->getDBname() );
 		$this->taskContext->setWikiDBW( $dbw );
 
-		foreach ( $this->sqlFiles as $file ) {
-			$this->debug( __METHOD__ . ": Populating database with {$file}" );
+		$dbw->multiQuery( $this->getSchemaSQL(), __METHOD__ );
+		$dbw->commit( __METHOD__ );
+
+		$this->waitUntilTableExistsOnSlave();
```

Comment:
As with creating the database, we want to poll the slave we're connected to to ensure the tables are ready before we move on to import data.

Reply:
_Ah, I hate waiting for slaves_ ― said Spartacus and escaped :)

But yeah, in this particular case it will address some issues and won't hurt a user, since they are waiting for wiki to be created anyway.